apps/examples/testcase/ta_tc/audio/utc/utc_audio_main.c

```diff
@@ -719,37 +726,38 @@ static void utc_audio_pcm_readi_n(void)
 static void utc_audio_pcm_writei_p(void)
 {
 	int fd;
-	ssize_t bytes;
 	int ret;
-	int total_frames_write = 0;
-	int remain;
 
-	fd = open(AUDIO_TEST_FILE, O_RDONLY);
-	TC_ASSERT_GEQ("pcm_writei", fd, 0);
+	char *buffer;
+	unsigned int size;
+	unsigned int bytes_per_frame;
+	int num_read;
 
 	/* use default config here */
 	g_pcm = pcm_open(0, 0, PCM_OUT, NULL);
-	TC_ASSERT_GT_CLEANUP("pcm_writei", pcm_is_ready(g_pcm), 0, close(fd));
-
-	bytes = lseek(fd, 0, SEEK_END);
-	lseek(fd, 0, SEEK_SET);
+	TC_ASSERT_GT("pcm_writei", pcm_is_ready(g_pcm), 0);
 
-	g_record_buffer = (char *)malloc(bytes);	//rate * bit * channels * duration
-	TC_ASSERT_NEQ_CLEANUP("pcm_readi_p", g_record_buffer, NULL, clean_all_data(fd));
+	size = pcm_get_buffer_size(g_pcm);//pcm_frames_to_bytes(g_pcm, pcm_get_buffer_size(g_pcm));
+	buffer = malloc(size);
+	TC_ASSERT_NEQ("pcm_writei", buffer, NULL);
```

Comment:
need cleanup for closing g_pcm? I don't understand why do you use global variable for each tc? It causes leak easily. Someone who want to modify these tc or review this should know whole seqeunces to prevent a memory leak. I think that is not good structure.

Reply:
@sunghan-chang Yes I agree with you but there is reason.
Compare audio fw and kernel, they are diffrent.
All these things except readi & writei, we should call sequentially.
For example, to test pcm_get_channels, we should follow below step if we don't use global
pcm_open -> pcm_set_config -> pcm_get_channels->pcm_close
I don't want make application as a calculating factorial, I just wanted to make it less function call per one utc.
We already discussed it but I'm writing this for my team member.